Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/wrapperUtils/UsageReporterOptions.java

```diff
@@ -0,0 +1,135 @@
+/*
+BSD 2-Clause License
+Copyright (c) 2018, Michael Doube, Richard Domander, Alessandro Felder
+All rights reserved.
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+package org.bonej.wrapperPlugins.wrapperUtils;
+
+import java.util.Random;
+
+import net.imagej.ImageJ;
+
+import org.scijava.ItemVisibility;
+import org.scijava.command.Command;
+import org.scijava.command.ContextCommand;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.widget.Button;
+
+/**
+ * Handles persistent settings such as user permission state.
+ * 
+ * Preferences are stored in their native format (long, int, boolean, etc.)
+ * 
+ * @author Michael Doube
+ * @author Richard Domander
+ *
+ */
+@Plugin(type = Command.class, menuPath = "Edit>Options>BoneJ Usage (Modern)")
+public class UsageReporterOptions extends ContextCommand {
+
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message1 = "Allow usage data collection?";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message2 = "BoneJ would like to collect data on";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message3 = "which plugins are being used, to direct development"; 
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message4 = "and promote BoneJ to funders.";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message5 = "If you agree to participate please check the box.";		
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String helpMessage = "For more information click Help.";
+	
+	@Parameter(label = "Opt in to usage data collection", description = "Can BoneJ send usage data?")
+  private boolean optIn = false;
+	
+	@Parameter(label = "Help")
+	private Button button;
```

Comment:
Help buttons aren't really necessary, since ImageJ now has the search bar, which looks for help from the forum and wiki.

Reply:
But I want to send to a particular URL on bonej.org - is it possible? Even to put a hyperlink in the message text or as a button action...? I want to make it as easy as possible (one-click-to-satisfaction) for users.

Reply:
Create a callback method for the button e.g.
```
@Parameter(label = "Help", callback = "openHelp")
private Button button;
```
and then with `PlatformService`
```
private void openHelp() {
  private final URL url = new URL("http://bonej.org");
  platformService.open(url);
}

I'm worried about sending people to `bonej.org` because that documentation is out of date with the plug-ins here.